go-controller/pkg/libovsdbops/db_object_types.go

```diff
@@ -8,21 +8,32 @@ const (
 const (
 	// owner types
 	EgressFirewallDNSOwnerType ownerType = "EgressFirewallDNS"
+	EgressFirewallOwnerType    ownerType = "EgressFirewall"
 	EgressQoSOwnerType         ownerType = "EgressQoS"
-	// only used for cleanup now, as the stale owner of network policy address sets
-	NetworkPolicyOwnerType   ownerType = "NetworkPolicy"
-	NetpolDefaultOwnerType   ownerType = "NetpolDefault"
-	PodSelectorOwnerType     ownerType = "PodSelector"
-	NamespaceOwnerType       ownerType = "Namespace"
-	HybridNodeRouteOwnerType ownerType = "HybridNodeRoute"
-	EgressIPOwnerType        ownerType = "EgressIP"
-	EgressServiceOwnerType   ownerType = "EgressService"
+	// NetworkPolicyOwnerType is deprecated for address sets, should only be used for sync.
+	// New owner of network policy address sets, is NetpolPeerASOwnerType.
+	NetworkPolicyOwnerType      ownerType = "NetworkPolicy"
+	NetpolDefaultOwnerType      ownerType = "NetpolDefault"
+	PodSelectorOwnerType        ownerType = "PodSelector"
+	NamespaceOwnerType          ownerType = "Namespace"
+	HybridNodeRouteOwnerType    ownerType = "HybridNodeRoute"
+	EgressIPOwnerType           ownerType = "EgressIP"
+	EgressServiceOwnerType      ownerType = "EgressService"
+	MulticastNamespaceOwnerType ownerType = "MulticastNS"
+	MulticastClusterOwnerType   ownerType = "MulticastCluster"
+	NetpolNodeOwnerType         ownerType = "NetpolNode"
+	NetpolNamespaceOwnerType    ownerType = "NetpolNamespace"
 
 	// owner extra IDs, make sure to define only 1 ExternalIDKey for every string value
 	PriorityKey           ExternalIDKey = "priority"
 	PolicyDirectionKey    ExternalIDKey = "direction"
 	GressIdxKey           ExternalIDKey = "gress-index"
 	AddressSetIPFamilyKey ExternalIDKey = "ip-family"
+	TypeKey               ExternalIDKey = "type"
+	IpKey                 ExternalIDKey = "ip"
+	PortPolicyIndexKey    ExternalIDKey = "port-policy-index"
+	IpBlockIndexKey       ExternalIDKey = "ip-block-index"
+	RuleIdx               ExternalIDKey = "rule-idx"
```

Comment:
it may be a bit confusing to use `rule-index` instead of `gress-index`, because every gress rule (whose index we are talking about) has a list or ports and a list of peers, and when you say `rule-index` it may not be obvious if it is gress/port/peer index

Reply:
I mentioned this because `rule` is the official documentation jargon for what we call `gress` here. The other stuff nobody would know what it is anyway because there are not well know terms for it. It's done in any case already, no point it dragging it much :P